--- a/obt/paths.c
+++ b/obt/paths.c
@@ -1,28 +1,33 @@
 static void find_uid_gid(uid_t *u, gid_t **g, guint *n)
 {
-    struct passwd *pw;
-    const gchar *name;
-    struct group *gr;
+    gid_t gmain;
+    unsigned int maininc;
+    int i;
 
     *u = getuid();
-    pw = getpwuid(*u);
-    name = pw->pw_name;
+    gmain = getgid();
 
-    *g = g_new(gid_t, *n=1);
-    (*g)[0] = getgid();
+    *n = getgroups(0, NULL);
+    if ((int)*n < 0) *n = 0;
+    *g = g_new(gid_t, *n + 1);
+    if (*n > 0)
+        *n = getgroups(*n, *g);
+    else
+        *n = 0;
 
-    while ((gr = getgrent())) {
-        if (gr->gr_gid != (*g)[0]) { /* skip the main group */
-            gchar **c;
-            for (c = gr->gr_mem; *c; ++c)
-                if (strcmp(*c, name) == 0) {
-                    *g = g_renew(gid_t, *g, ++(*n)); /* save the group */
-                    (*g)[*n-1] = gr->gr_gid;
-                    break;
-                }
+    /* Check if the effective group ID of the calling process is already
+       included in the returned list. Add it otherwise. */
+    maininc = 0;
+    for (i = 0; i < (int)*n; i++) {
+        if ((*g)[i] == gmain) {
+            maininc = 1;
+            break;
         }
     }
-    endgrent();
+    if (!maininc) {
+        (*g)[*n] = gmain;
+        (*n)++;
+    }
 
     qsort(*g, *n, sizeof(gid_t), gid_cmp);
 }
